tensorrt_llm/_torch/auto_deploy/transformations/library/sharding.py

```diff
@@ -103,15 +87,25 @@ class TPShardingInfo(ShardingTransformInfo):
     dist_op: Optional[Literal["all_reduce", "all_gather"]] = None
     min_local_shape: int = 1
 
-    def apply(self, gm: GraphModule, node: Node) -> None:
-        """Apply TP sharding transformation to the graph module."""
-
-        # first, verify whether the config is valid
+    def validate(self, gm: GraphModule = None, node: Node = None) -> bool:
```

Comment:
In the Pydantic world this function can easily be written as a "model_validator". Here is an example from the documentation: https://docs.pydantic.dev/latest/concepts/validators/#model-validators

Some as my other comment. We can keep the current version and refine in future iterations